Bahc4 478 pixmart dev

MigleMat opened the pull request:
This pull request introduces the full initial version of the Pixmart project from the `BAHC4-478-pixmart-dev` branch into `main`.

MykolaKushka commented:
You have [3] more attempt to finish your diploma project

### 1. General
- [x] 1.1 ❤️ Gulp is used as a build tool
- [x] 1.2 ❤️ Public and node_modules folders should not be in the GitHub repository
- [x] 1.3 ❤️ The project builds correctly
- [x] 1.4 ❤️ No errors (error) in the console during the build process
- [x] 1.5 💛 No warnings (warning) in the console during the build process
- [x] 1.6 ❤️ File names and extensions are lowercase and contain no spaces
- [x] 1.7 ❤️ Uniform code formatting
- [ ] 1.8 ❤️ Layout matches design (Perfect Pixel)

The gradient in .hero__container needs to be checked. In Figma, for example, the Watch About Us button is clearly visible, but in the implementation its white background blends into the container background.

![fe83715d-7872-4335-b891-074d24814836](https://github.com/user-attachments/assets/9336dc05-fb90-4d04-af02-2492fcdd87a7)

The distance for mobile in the Categories section needs to be checked, and also provide a bottom space for the horizontal scroll so that it does not touch the content directly. The same applies to other sections that have horizontal scrolling.

![b029f826-9cd9-4f03-9b20-3db7921e4c8e](https://github.com/user-attachments/assets/69f2eda4-5210-4511-824c-28895c3212cd)

On mobile, the Plus Membership block overlaps the upper section on hover, revealing the underlying section. It’s recommended to either add a white background to this block to cover the lower section or ensure consistent spacing between the Pricing Plan blocks for mobile – even if this differs from Figma. The first option looks simpler and doesn’t require design adjustments.

![5738328b-f15f-4994-901e-309547ea5c7b](https://github.com/user-attachments/assets/e0576553-6b00-435f-959b-2fef733a7cea)

The size of the first .article-card__picture for mobile needs to be checked.
The alignment of the content in footer__copyright on mobile needs to be checked.

![06c6071b-4a31-4afd-9940-4fc93fdc13e4](https://github.com/user-attachments/assets/e647f3e4-1dd6-4c92-87a4-77c1336ff537)

The pricing plans are cut off at the top on tablet screens.
The .faq-section__list-item on desktop should not jump in width depending on the content.
The spacing between .accordion__question and .accordion__answer needs to be checked.

![899f084e-5f9f-4fc5-95a0-2766109394c6](https://github.com/user-attachments/assets/ad789ead-831f-41ec-b9d3-26a195568329)

The dropdown menu should completely cover the content underneath.
The background positioning for .catalog-banner__container does not match the design on mobile, and the wave overlaps the sorting-controls__dropdown
The .catalog-card__overlay overlaps the SOLD OUT label, so the color does not match the design.
There is no sense in adding hover states for .catalog-card__price on cards marked as 'Sold Out'.
The size of the .sorting-controls__dropdown for tablets needs to be checked.
The spacing for .sorting-controls__tag needs to be checked.
The bottom row of .catalog__list-item for tablets and desktops should be covered by a gradient.
The borders for .sidebar__categories-title and .sidebar__latest-articles-group on mobile should be 100% width
The top margin of the footer in Blog Details on mobile needs to be checked.
In .contacts__list-item, the icons should be vertically aligned.
The .contacts__map should have rounded corners.
The .customer-form__select should not have a green border and the default background.
Check the styles of checkboxes and radio buttons in .catalog-filters__list in both default and selected states.

- [x] 1.9 ❤️ Cross-browser compatibility
- [x] 1.10 ❤️ Handles content overflow gracefully

### 2. GIT
- [x] 2.1 ❤️ The site has a GitHub repository
- [ ] 2.2 ❤️ No system files like `.DS_Store` or `Thumbs.db` in the repository

The file package-lock.json is present in the repository. It should be removed.

- [x] 2.3 ❤️ No commits in the target branch (usually `main`)
- [x] 2.4 💛 Commit messages are descriptive
- [x] 2.5 💛 Each commit is a logical unit
- [x] 2.6 💚 Commits are atomic
- [ ] 2.7 💚 Commit messages are no longer than 60 characters

There are commits with titles longer than 60  characters

### 3. Nunjucks
- [x] 3.1 ❤️ Project is structured using Nunjucks templates
- [x] 3.2 💛 Loops, variables and dynamic data are used in Nunjucks

### 4. HTML
- [x] 4.1 ❤️ Document begins with `<!DOCTYPE HTML>`
- [x] 4.2 ❤️ Each page has the `lang` attribute set
- [x] 4.3 ❤️ Encoding is specified (UTF-8)
- [x] 4.4 ❤️ `<title>` tag is set on all pages
- [x] 4.5 ❤️ All elements are marked up and styled
- [x] 4.6 ❤️ No gross markup errors
- [x] 4.7 ❤️ Semantic markup is used
- [ ] 4.8 ❤️ Layout is valid (passes W3C validator)

The blog page does not pass validation. The Products page also does not pass validation, and there are CSS errors as well. It’s worth checking with https://validator.nu/.

- [x] 4.9 ❤️ All pages are linked; no broken links
- [x] 4.10 ❤️ Each page has an `<h1>` different from the page title
- [x] 4.11 💛 Heading tree structure is maintained
- [x] 4.12 💛 External links use `target="_blank"`
- [x] 4.13 💛 Minimum possible number of HTML elements
- [ ] 4.14 💛 Logo on internal pages links to the home page (not on home)

The logo links to the home page even on the home page.

- [x] 4.15 💚 `<meta name="description">` is set and unique

### 5. BEM
- [x] 5.1 ❤️ Independent components are defined as blocks
- [x] 5.2 ❤️ Subordinate parts are correctly defined as elements
- [x] 5.3 ❤️ BEM element is not used outside its block
- [x] 5.4 ❤️ Modifiers correctly describe states or variations
- [x] 5.5 ❤️ Modifiers must not include colours, numbers or tag names
- [x] 5.6 ❤️ Modifiers are not the only class on a tag
- [x] 5.7 ❤️ BEM syntax rules are followed
- [ ] 5.8 ❤️ Block, element and modifier names are meaningful

The purpose of the classes layout-container--small and layout-container--medium is unclear.

- [x] 5.9 ❤️ Classes are named correctly (lowercase, no transliteration, no presentational names)

### 6. Images
- [x] 6.1 ❤️ Correct image format (WEBP / SVG)
- [x] 6.2 ❤️ Images are extracted without style filters from the design
- [x] 6.3 ❤️ Correct image insertion method (`<img>` vs `background-image`)
- [ ] 6.4 ❤️ Dimensions are specified for all images

Should be checked

- [x] 6.5 ❤️ All images have `alt` text
- [x] 6.6 ❤️ Vector SVG sprite is used
- [ ] 6.7 💚 Favicon icons display correctly

### 7. Forms
- [x] 7.1 ❤️ Appropriate `type` attributes are set for inputs
- [x] 7.2 ❤️ Every form element has a `<label>`
- [x] 7.3 ❤️ Basic HTML form validation is implemented
- [x] 7.4 💛 Resizing `<textarea>` does not break the layout

### 8. Preprocessor
- [x] 8.1 ❤️ SCSS preprocessor is used
- [x] 8.2 ❤️ Styles are split into components
- [x] 8.3 ❤️ Variables (SCSS / CSS custom properties) are used for colours and fonts
- [x] 8.4 💛 Mixins are used where appropriate
- [x] 8.5 💛 No magic numbers

### 9. CSS
- [x] 9.1 ❤️ Single CSS file (plus optional Normalize.css)
- [x] 9.2 ❤️ All design fonts are connected
- [x] 9.3 ❤️ Fallback fonts/families are provided
- [x] 9.4 ❤️ `!important` is not used
- [x] 9.5 ❤️ `#id` is not used for styling
- [x] 9.6 ❤️ Consistent units for sizing and positioning
- [x] 9.7 ❤️ Appropriate properties are used for animations (e.g. `transform`)
- [ ] 9.8 ❤️ All element states from the style guide are implemented

Different elements jump on hover because a border is added, changing their height and width. This should be avoided. This issue is present in elements like faq-section__list-item, footer__newsletter-input, search__input, catalog__button-link, related-articles__list-item, as well as in other elements like input, textarea, and button.

- [x] 9.9 ❤️ Links respond to `:hover`, `:active`, `:focus`
- [x] 9.10 💛 Selector nesting is no deeper than two levels
- [x] 9.11 💛 Tags are not styled directly (except `body`, `ul`, `a`, `img`)
- [x] 9.12 💛 Elements with background images have matching background colour
- [x] 9.13 💛 Colours are in a consistent format (hex / rgba)
- [ ] 9.14 💛 Interactions do not change document flow 

Described in 9.8

- [x] 9.15 💚 Normalize.css is included

### 10. Layout
- [x] 10.1 ❤️ Flex or Grid is used for layouts
- [ ] 10.2 ❤️ Responsive adaptive layout is implemented 
Check 1.8
- [x] 10.3 ❤️ No horizontal scroll on the full page
- [x] 10.4 ❤️ Burger menu works on mobile
- [ ] 10.5 ❤️ Filter panel opens correctly on mobile

Needs to be checked
![image](https://github.com/user-attachments/assets/63d7a13b-5f5d-4e4a-b3c7-e8f0477ba3b2)

- [x] 10.6 ❤️ Horizontal scroll enabled for specified blocks on mobile
- [ ] 10.7 ❤️ Breakpoints are well chosen

The website should match the design and display correctly on all screens starting from a minimum width of 440px without breaking.

- [ ] 10.8 ❤️ Consistent spacing and alignment

Described in 1.8 

- [x] 10.9 ❤️ Interactive elements are accessible on mobile

MigleMat commented:
1. _The gradient in .hero__container needs to be checked. In Figma, for example, the Watch About Us button is clearly visible, but in the implementation its white background blends into the container background._
-FIXED. Before I chose wrong gradient direction, changed from `from top to bottom` to `from left to right`

2. _The distance for mobile in the Categories section needs to be checked, and also provide a bottom space for the horizontal scroll so that it does not touch the content directly. The same applies to other sections that have horizontal scrolling._
-FIXED. Added bottom space on all scrolable elements and removed unnecessary left spacing on categories section

3._On mobile, the Plus Membership block overlaps the upper section on hover, revealing the underlying section. It’s recommended to either add a white background to this block to cover the lower section or ensure consistent spacing between the Pricing Plan blocks for mobile – even if this differs from Figma. The first option looks simpler and doesn’t require design adjustments._ 
-FIXED. Based on your comment, I added white backgrond. Card on hover still overlaps, but visually look much better. I hope it's okey. 

4._The size of the first .article-card__picture for mobile needs to be checked._ 
FXED

5._The alignment of the content in footer__copyright on mobile needs to be checked._
FIXED

6_.The pricing plans are cut off at the top on tablet screens._ 
FIXED

7._The .faq-section__list-item on desktop should not jump in width depending on the content._ 
FIXED

8._The spacing between .accordion__question and .accordion__answer needs to be checked._ 
Hm... here I have gap:16px. same as in Figma design. WHat is wrong with my spacing here? 

9. _The dropdown menu should completely cover the content underneath._ 
FIXED

10. _The background positioning for .catalog-banner__container does not match the design on mobile, and the wave overlaps the sorting-controls__dropdown_
FIXED

11. _The .catalog-card__overlay overlaps the SOLD OUT label, so the color does not match the design._
FIXED. Used z-index on the label. 

12. _There is no sense in adding hover states for .catalog-card__price on cards marked as 'Sold Out'._
I kind of agree, but this is what we have in our Figma desing. Shall I still change or keep it as it is? 

![Screenshot 2025-06-09 152046](https://github.com/user-attachments/assets/9c3ae7fb-c73b-4aa4-8e08-937b2714f978)

13_. The size of the .sorting-controls__dropdown for tablets needs to be checked._
 and _The spacing for .sorting-controls__tag needs to be checked._
FIXED

14. The bottom row of .catalog__list-item for tablets and desktops should be covered by a gradient. 
FIXED

MykolaKushka commented:
![image](https://github.com/user-attachments/assets/0c732f22-cf8a-40d6-b3e0-8f48a2525195)

8.The spacing between .accordion__question and .accordion__answer needs to be checked.
_Hm... here I have gap:16px. same as in Figma design. WHat is wrong with my spacing here?_

In Figma 16px is between bottom of arrow and answer text.

MykolaKushka commented:
There is no sense in adding hover states for .catalog-card__price on cards marked as 'Sold Out'.
_I kind of agree, but this is what we have in our Figma desing. Shall I still change or keep it as it is?_

Well, lets leave as it is on Figma then)

MigleMat commented:
- Burger menu navigation fixed (not scrolling, buttons do not jump when click)
- Filter menu fixed (added X, inrease X button size, submit button given space) 
-Hover for checkboxes when selected - fixed 
-Made layout container wider - better for wider screens 
-fixed  wrong padding on some sections
-validation for CSS - fixed

MykolaKushka commented:
You have [1] more attempt to finish your diploma project

### 1. General
- [ ] 1.8 ❤️ Layout matches design (Perfect Pixel)

_The spacing between .accordion__question and .accordion__answer needs to be checked._
_Rev 2. The spacing is still wrong._
Rev 3. It is done.

_The dropdown menu should completely cover the content underneath._
_Rev 2. This much better now but could you prevent scrolling content under menu when it's opened?_ 
Rev 3. It needs changes

_The size of the .sorting-controls__dropdown for tablets needs to be checked._
_Rev 2. Still needs to be checked for different devices_
Rev 3. It is done.

_Check the styles of checkboxes and radio buttons in .catalog-filters__list in both default and selected states._
_Rev 2. Still needs to be checked for checkboxes (hover for default and for checked)_
Rev 3. It is done.


### 2. GIT

- [ ] 2.7 💚 Commit messages are no longer than 60 characters

_There are commits with titles longer than 60  characters_
_Rev 2. It wasn't changed_
Rev 3. It wasn't changed

### 4. HTML
- [x] 4.8 ❤️ Layout is valid (passes W3C validator)

_The blog page does not pass validation. The Products page also does not pass validation, and there are CSS errors as well. It’s worth checking with https://validator.nu/._
_Rev 2. CSS still has errors._
Rev 3. It is done.

### 9. CSS

- [x] 9.8 ❤️ All element states from the style guide are implemented

_Different elements jump on hover because a border is added, changing their height and width. This should be avoided. This issue is present in elements like faq-section__list-item, footer__newsletter-input, search__input, catalog__button-link, related-articles__list-item, as well as in other elements like input, textarea, and button._

Rev 2. Check also active menu item on mobile menu, Related Articles item on hover
Rev 3. It is done.

- [x] 9.14 💛 Interactions do not change document flow 

Described in 9.8

### 10. Layout

- [ ] 10.2 ❤️ Responsive adaptive layout is implemented 
Check 1.8

- [ ] 10.7 ❤️ Breakpoints are well chosen

_The website should match the design and display correctly on all screens starting from a minimum width of 440px without breaking._

_Rev 2. Still need to be checked. For example when device is 768px wide it looks so._
Rev 3. It wasn't changed

- [ ] 10.8 ❤️ Consistent spacing and alignment

_Described in 1.8_ 
Rev 3. It is based now on 10.7

MykolaKushka commented:
Congratulations!!! Well done. All obligatory requirements were changed and pushed. Good job!  Your diploma is approved.

MigleMat commented:
Thank you so much!
